src/posthog-core.js

```diff
@@ -596,6 +596,8 @@ PostHogLib.prototype.capture = addOptOutCheckPostHogLib(function (event_name, pr
     }
     if (typeof callback !== 'function') {
         callback = __NOOP
+    } else {
+        window.console.warn('WARNING! Calling posthog.capture with a callback is deprecated and will be removed soon!')
```

Comment:
One of the other issues is to make this library run [in a SSR environment](https://github.com/PostHog/posthog-js/issues/114) where `typeof window === 'undefined'`. I'd just skip `window.` here since `console` will (probably?) be a "superglobal" in every context this library will run in. For sure in the browser, in nodejs and in every vm-style/virtualized JS setup I've seen.

Somewhat related, a lot of the code in this library links to [this console reimplementation](https://github.com/PostHog/posthog-js/blob/d7264c46a2d30923d1e5cd75e23f9c2186e10387/src/utils.js#L36), which seems to be a error logger that defaults to not logging (`debug: false`). Not sure if I'd use it in this case. 

This "quiet" logger might make sense in a CLI app where you want less noise output to your users, but that's not the target user of this library. We might merge all JS implementations one day, so I'm not sure if we should keep this logic intact or not... Out of scope, in any case.

Reply:
So the motivation for using window.console.warn here was existing code in the same file that does the same (for deprecated methods).

I'd love to get rid of the console utils and likely will do it one day - it adds noise.

Re SSR: one thing I have yet to understand is if 'running' in a SSR environment means being able to include the module or also init-ing it and doing other things. Very different in terms of expected behavior.

Reply:
Hmm, indeed, it's the least bad solution right now to make sure people see the deprecation. 

For SSR, it's minimally the expectation that this will work:

```fish
(env) Mariuss-MacBook-Pro ~/P/P/posthog-js/dist [capture-hook-pageviews] (ruby 2.6.3p62) $ node 
Welcome to Node.js v14.5.0.
Type ".help" for more information.
> const posthog = require('./module')
Uncaught ReferenceError: window is not defined
    at Object.<anonymous> (/Users/marius/Projects/PostHog/posthog-js/dist/module.js:619:21)
    at Module._compile (internal/modules/cjs/loader.js:1201:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1221:10)
    at Module.load (internal/modules/cjs/loader.js:1050:32)
    at Function.Module._load (internal/modules/cjs/loader.js:938:14)
    at Module.require (internal/modules/cjs/loader.js:1090:19)
    at require (internal/modules/cjs/helpers.js:75:18)
```

Normally you wouldn't want to do any HTTP requests while SSR'ing. The aim is to render the page as quickly as possible. Background requests are possible though, but I'd personally not init posthog server side as all the double events would get annoying.

Reply:
Thanks for this - I will aim to solve the requiring bit today - the overall window usage is a much harder issue though! :)